share/availability/discovery/set.go

```diff
@@ -44,8 +44,11 @@ func (ps *limitedSet) Size() int {
 }
 
 // Add attempts to add the given peer into the set.
-func (ps *limitedSet) Add(p peer.ID) {
+func (ps *limitedSet) Add(p peer.ID) (added bool) {
 	ps.lk.Lock()
+	if _, ok := ps.ps[p]; ok {
+		return false
+	}
```

Comment:
Everywhere we do Add, we also do Contains before, so returning bool is unnecessary

Reply:
Contains is happening outside of mutex lock scope, so it is not a safe to assume peer was not added inbetween operations

Reply:
Yeah, so let remove the contains as unnecessary check